src/MosparoIntegration/Helper/ConfigHelper.php

```diff
@@ -258,9 +259,60 @@ public function disableModule($moduleKey)
         $moduleIndex = array_search($moduleKey, $this->config['modules']);
         if ($moduleIndex !== false) {
             unset($this->config['modules'][$moduleIndex]);
+            //Also remove module settings on disable
+            unset($this->config['modules-settings'][$moduleKey]);
         }
     }
 
+    public function getTypedValue($value, $type = 'text') {
+        switch ($type) {
+        case "boolean":
+            return filter_var($value, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE);
+        case "number":
+            return filter_var($value, FILTER_VALIDATE_INT, $options = array('options' => array('default' => 0)));
+        case "string":
+        case "text":
+        default:
+            break;
+        }
+        return strval($value);
+    }
+
+    public function loadModuleConfiguration(AbstractModule $module) {
+        $isMultisite = is_multisite();
+        $isNetworkAdminPage = is_network_admin();
```

Comment:
These two variables are not used and could be removed.

But I think we have to do something with them... Because right now, we're loading the module configuration from the active place, so in the network administration, we load the module configurationfrom the network while in the normal page (in the frontend) we're loading the module configuration from the local WordPress website.

Maybe we should check where the module was enabled (Network or Website) and load the configuration from there.

Reply:
Indeed, the configuration was not loaded properly when network-activated.
I pushed a little fix for that (7501703f962afd7db559ec0c65192f3fc4075eb4).
I think there is a priority problem regarding module though: the "wordpress way" is to give priority to single site options when a plugin is activated both network-wide and on single site.
For mosparo modules, the logic is reversed and network settings take priority.
This isn't really a problem because settings are not accessible on single site if the module was network-activated but a little rework of network support is needed imo.